### ISSUE TITLE
Add Webservice Request: SmarterMail

### ISSUE THREAD

**Issue by lewismll** [Noitidart/MailtoWebmails#21]
I was trying to install a webmail service from the options panel of MailtoWebesrvices but the one I was looking for was not found. I was looking for SmarterMail (www.smartertools.com)

**Comment by Noitidart** [comment]
Ill look into it :)

**Comment by lewismll** [comment]
Thanks, if you need a demo account just let me know, we have our own copy of the e-mail service.

**Comment by Noitidart** [comment]
Hey @lewismll do you know the compose url? I couldnt figure it out.

**Comment by lewismll** [comment]
New EMAIL:
https://mail.DOMAIN.com/Main/frmCompose.aspx?popup=true

Reply to EMAIL:
https://mail.DOMAIN.com/Main/frmCompose.aspx?action=reply&folder=Inbox&uid=1022&user=USERID&mapped=False&fromPreview=true&rowNumber=0&popup=true

Forward EMAIL:
https://mail.DOMAIN.com/Main/frmCompose.aspx?action=forward&folder=Inbox&uid=1022&user=USERID&mapped=False&fromPreview=true&rowNumber=0&popup=true

**Comment by Noitidart** [comment]
Please beta test submission of the url :)

1. Install addon GitHub Extension Installer: https://addons.mozilla.org/en-US/firefox/addon/github-extension-installer/
* Go to latest version of MailtoWebmails on GitHub repository: https://github.com/Noitidart/MailtoWebmails
* Click on "Add to Firefox", this install this version to your Firefox
* Then open up the prefs page, or go to about:mailto
* Then submit zimbra :)

**Comment by lewismll** [comment]
This is what I get when clicking on a mailto:
https://mail.DOMAIN.com/Main/frmCompose.aspx?to=training@somewhere.com&popup=true

**Comment by Noitidart** [comment]
Ah ok i think thats it then. If you install v2 from github you can plug it in and test it out right away. :)

**Comment by Noitidart** [comment]
the url you would put into the url field would be this:
`https://mail.DOMAIN.com/Main/frmCompose.aspx?to=%s&popup=true`

**Comment by Noitidart** [comment]
Hi htere @lewismll version 2 addon review is taking some time, can you hook me up with a demo account so i can test this out.

**Comment by lewismll** [comment]
Here you go:

smtp:mail.segotest.com
web: https://mail.segosystems.com
U: tester@segotest.com
P" T3st3r20!

Active until October 5 @ 13:00

**Comment by Noitidart** [comment]
Ahh so this is a thing like RoundCube!! Everyone has their custom domain. Ok let me set up a template just like roundcube :)

**Comment by lewismll** [comment]
Also, V2 seems to not be compatible with PaleMoon browser http://www.palemoon.org/ (FFOX alternate), so I can't assist with the test, the only other browser I have is IE.

**Comment by Noitidart** [comment]
Oh yikes supporting palemoon seems like a doozy, they seem to have stopped updating their codebase since Firefox 24 and I use the new technologies :( From a quick look

I found a bug which im working on right now so didnt dive too deep into it. Will do later.

**Comment by lewismll** [comment]
Yes, they wanted to continue the old FFOX line of code when FFOX decided to change and use the chrome engine.

No problem; all our users access webmail using Chrome, I'm just the odd-duck!

**Comment by Noitidart** [comment]
Ooo yuck they should use Firefox :(

**Comment by lewismll** [comment]
I was just about to say "& Firefox" :)

**Comment by Noitidart** [comment]
Hahaha. I edited in to make sure you know you're the smart duck not the odd duck to not use Chrome :P
They give you a product for "free". They are really making tons of cash off of you and when you have feedback Google won't listen. So that free is actually super strings attached and they aren't up front about it. Things are allways opt-out, by default the force it on you and you dont know you're giving away till its too late lol

**Comment by Noitidart** [comment]
Too late == junk mail in your physical mail box at home as one of the first side effects

**Comment by lewismll** [comment]
Agree 100%, I don't like how they do business. So if we need to switch the remaining users over to FFOX for this to work - I have noooo problems! lol

**Comment by Noitidart** [comment]
Haha ill definitely look into Palemoon support after fixing some bugs too, as it shouldn't be too hard, I know Firefox forced some stuff on users like Australis so I undrstand.

Whats your reason for Palemoon btw?

**Comment by lewismll** [comment]
It was mainly because FFOX was changing things so often that many of my add-on's broke and I had to wait until the update was made or Mozilla fixed the bug. So I went with something with less updates but compatible with most of the add-on's at FFOX. Have not run into too many problems with add-on's.

Sometimes the newest shinny object is not always the best :)

**Comment by Noitidart** [comment]
Ah haha

**Comment by Noitidart** [comment]
Ok finally I fixed a bug i had server side where users could not submit names with name or descriptions including a double quote.

So if you install v2.0 on firefox you'll see "SmarterMail Template" for right now i set it up to use your domain and its not right:

So clicking on a mailto link pops open this url:

    https://mail.segosystems.com/Main/frmCompose.aspx?to=mailto%3Anoitidart%40gmail.com&popup=true

So we dont have to use my addon to test, can you please help me find a url where we can navigate that will open the compose url with the to line filed in. The url should process `mailto:blah@blah.com` and convert that to `blah@blah.com`, thats how mailto protocol handler works in firefox and how all the other url templates work, they accept the `mailto:` prefix and they take it out.

This is the bad page it would go to:

![](http://i.imgur.com/2xI6Kb4.png)

**Comment by Noitidart** [comment]
Oh waiiiit I wasnt logged in. So thats a bug on their smartermail side, they should show a "you are not logged in" message if it loads that page while user is not logged in.

So this is what I see:

![](http://i.imgur.com/0bLWOGu.png)

So we still see the issue where it is prefixed with `mailto:` so we have to find a smartermail url that removes that prefix. Need your help there please, we may have to contact smartermail support.

**Comment by Noitidart** [comment]
Also the SmarterMail Template will be added as public and wont have seago as its domain it will be YOUR_DOMAIN_HERE as its a template. Do you want me to add a Seago one as well? Publically visible?

**Comment by Noitidart** [comment]
Btw what i did (other then the bug fixing) is nothing only i can do. Its a public interface, you just need to install v2.0 and you can add and edit whatever you want :)

So what I did was using the v2 interface i created a template, and then i created one for sego:

I approved both for public visibility. Template for sure we want public. And the sego one we want public too right? Or do you want me to delete that so its not public? And your users can come in and modify the template?

![](http://i.imgur.com/jsyROTW.png)

**Comment by Noitidart** [comment]
Your update came in and changed the sego icon from the white to this:

![](http://i.imgur.com/NNt5bRG.png)

are you sure you want that? it doesnt match the rest of the logos in the mailtowebmails. its just the white part we want. but up to you.

**Comment by lewismll** [comment]
You're right, I think the other way was better - thanks for checking & asking.

**Comment by Noitidart** [comment]
Thanks I appreicate you trying to match the mailtwoebmails style :)

**Comment by sgo441** [comment]
Here is the correct URL which removes the mailto function for SmarterMail. Hopefully this is what you are needing.

http://mail.segosystems.com/Main/frmCompose.aspx?to=aaron.galindo@teamsewell.com&popup=true

**Comment by Noitidart** [comment]
Thanks @sgo441 that's the same url we are using. However the `aaron.galindo@teamsewell.com` is repalced with a `%s` in our template.

Now when a user clicks on a mailto link that `%s` gets replaced with `mailto:aaron.galindo@teamsewell.com` that's just how the protocol was standardized all browsers and apps do that. The mailto: is to be stripped out server side, as does hotmail, ymail, gmail, etc.

Thanks very much though for your input it's great to see a team effort :)

If you can get in touch with smartermail support with this info:

>
 I was using this URL for smartermail:

    https://mail.USER_DOMAIN_HERE.com/Main/frmCompose.aspx?to=%s&popup=true

The `%s` gets replaced with the mailto link a user clicked on. For example if the link is `mailto:noitidart@gmail.com` it will be come:

    https://mail.USER_DOMAIN_HERE.com/Main/frmCompose.aspx?to=mailto%3Anoitidart%40gmail.com&popup=true

Notice how the `%s` is now `mailto%3Anoitidart%40gmail.com` (that is url encoded for `mailto:noitidart@gmail.com`)

It works ok, but the problem is your server is not removing the "mailto:" prefix. All other webmail clients, hotmail, outlook, gmail, fastmail, comcast, zimbra, etc remove the mailto: prefix. May I request you to please make this correction so the `mailto:` prefix is removed. Please see attached screenshot.

Screenshots to attach:
![mailto prefix gets included](https://cloud.githubusercontent.com/assets/6372489/10296806/17e9209c-6b81-11e5-869b-7bb92b590d2b.png)

**Comment by sgo441** [comment]
I see what you mean. It's interesting because even when you click on an email address inside of SmarterMail, it still applies the mailto. It's only when you hoover over an email address then select Email that SmarterMail automatically removes the mailto feature. Hopefully they can implement this for other URLs.

![2015-10-05_18-56-31](https://cloud.githubusercontent.com/assets/12103008/10296902/ca894306-6b92-11e5-8e4f-936e0034997d.png)

**Comment by Noitidart** [comment]
I emailed smartermail support earlier with the same message. They responded but asked me to open a ticket which I can't do because I'm not a customer. I emailed them back again explaining, I haven't heard back yet.

May I forward this email to you? I'll be forwarding from noitidart@gmail.com

**Comment by lewismll** [comment]
Just a quick update: We spoke with the SmarterTools team and they are going to fix it!

***"The developers have added this onto their feature request list and we will be working to implement this further. This will likely fall into our next minor release which is scheduled for about 3-4 weeks out."***

I will let you know when it comes through so we can further test.

**Comment by Noitidart** [comment]
Superb! Thanks team!

**Comment by ghost** [comment]
@Noitidart if you need help getting support for Smartermail / Smartertools then give me a bell. i have multiple paid memberships so can raise requests etc quite easily. 

I just found your add in and will be letting SM know about it as well as telling clients. This closes the loop for us as a mail provider - letting mail-to links work like a native app.

Thank you.
http://mail.postbox.net.nz/Main/frmCompose.aspx?to=%s&popup=true

BTW - is there an option to handle subject, cc, bcc ?

**Comment by Noitidart** [comment]
Thank you @oitconz !! When clicking on `mailto:` href links, the protocol handler only allows `to:`. :( Subject, CC, and BCC are not supported. But I can inject a content script into each page and override it, it's much less performant, and room for error in case mailto links are dynamically added and I miss it with a MutationObserver. But I might hook it up this content script method in future, because we are having major problems with Microsoft/Outlook working with us.

**Comment by ghost** [comment]
Thanks for the reply.  I was  just curious as some Web formats do allow subject etc from the anchor / href.

Regards,
Shane HollisOutsourced Information Technology LtdM 022 626 5084

-------- Original message --------

**Comment by hdijkema** [comment]
Does anyone know how to insert a body?
